### PR TITLE
Enable contextMenuBuilder in the absence of selectionControls

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -448,7 +448,7 @@ class TextSelectionOverlay {
   void showToolbar() {
     _updateSelectionOverlay();
 
-    if (selectionControls is! TextSelectionHandleControls) {
+    if (selectionControls != null && selectionControls is! TextSelectionHandleControls) {
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/text_selection.dart:451]: Yeah this was my mistake when I wrote this, thanks for the fix.

File: packages/flutter/test/widgets/text_selection_test.dart [hunk 2]
```diff
@@ -1781,6 +1781,51 @@ void main() {
...
+        ),
+      ),
+    );
+
+    final Offset position = textOffsetToPosition(tester, 0);
+    expect(find.byType(Placeholder), findsNothing);
+    await tester.tapAt(position, buttons: kSecondaryMouseButton, kind: PointerDeviceKind.mouse);
+    await tester.pump();
+    expect(find.byType(Placeholder), findsOneWidget);
+  }, skip: kIsWeb); // On web, we use native context menus for text fields.
```
Comment (justinmc) [inline packages/flutter/test/widgets/text_selection_test.dart:1828]: There's an analyzer error, I think you need `[intended]`?

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/test/widgets/text_selection_test.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/test/widgets/text_selection_test.dart]: ah yes thanks

**Pull request merged**